frontend/src/components/AbstractModal.tsx

```diff
@@ -0,0 +1,66 @@
+import { Dialog, DialogContent, DialogTitle, DialogActions, Typography, makeStyles, IconButton } from '@material-ui/core';
+import CloseIcon from '@material-ui/icons/Close';
+import * as React from 'react';
+import Button from './Button';
+
+
+export type modalType =  {
```

Comment:
As a convention, type names should be PascalCase
```suggestion
export type ModalType =  {
```